src/Build/BackEnd/Client/MSBuildClient.cs

```diff
@@ -238,7 +243,24 @@ private void SupportVT100()
             }
         }
 
-        private void SendCancelCommand(NamedPipeClientStream nodeStream) => throw new NotImplementedException();
+        private bool TrySendPacket(Func<INodePacket> packetResolver)
+        {
+            INodePacket? packet = null;
+            try
+            {
+                packet = packetResolver();
+                WritePacket(_nodeStream, packet);
+                CommunicationsUtilities.Trace($"Command packet of type '{packet.Type}' sent...");
+            }
+            catch (Exception ex)
+            {
+                CommunicationsUtilities.Trace($"Failed to send command packet of type '{packet?.Type.ToString() ?? "Unknown"}' to server: {0}", ex);
```

Comment:
If packet is null, I think this calls ToString on null. packet?.Type?.ToString()?

Reply:
I don't think so. If packet is `null`, both `Type` getter and `ToString` are not invoked.